core/services/functions/orm.go

```diff
@@ -54,7 +60,14 @@ func (o *orm) CreateRequest(requestID RequestID, receivedAt time.Time, requestTx
 		INSERT INTO ocr2dr_requests (request_id, contract_address, received_at, request_tx_hash, state)
 		VALUES ($1,$2,$3,$4,$5);
 	`
-	return o.q.WithOpts(qopts...).ExecQ(stmt, requestID, o.contractAddress, receivedAt, requestTxHash, IN_PROGRESS)
+	err := o.q.WithOpts(qopts...).ExecQ(stmt, requestID, o.contractAddress, receivedAt, requestTxHash, IN_PROGRESS)
+	var pgErr *pgconn.PgError
```

Comment:
This approach seems very fragile to me.
Why not do `insert into ... on conflict (request_id) do nothing`, then you check affected rows count returned from Exec(). If the count is zero, then you had a conflict. 
@reductionista do you agree?

Reply:
But then if Exec fails for another reason (e.g. context expired) the row count will be zero as well, right? I want to differentiate a duplicate ID specifically. Another way is to have a transaction with a GET first and then INSERT later.

Reply:
The err will not be nil, this is how you differentiate.

Reply:
```golang
func (o *orm) CreateRequest(requestID RequestID, receivedAt time.Time, requestTxHash *common.Hash, qopts ...pg.QOpt) error {
	stmt := `
		INSERT INTO ocr2dr_requests (request_id, contract_address, received_at, request_tx_hash, state)
		VALUES ($1,$2,$3,$4,$5) ON CONFLICT (request_id) DO NOTHING;
	`
	result, err := o.q.WithOpts(qopts...).Exec(stmt, requestID, o.contractAddress, receivedAt, requestTxHash, IN_PROGRESS)
	if err != nil {
		return err
	}
	nrows, err := result.RowsAffected()
	if err != nil {
		return err
	}
	if nrows == 0 {
		return ErrDuplicateRequestID
	}
	return nil
}
```